src/measure_theory/measure/haar.lean

```diff
@@ -644,6 +644,55 @@ begin
   exact regular.smul ctop,
 end
 
+/-- **Steinhaus Theorem** In any locally compact group `G` with a haar measure `μ`, for any
+  measurable set `E` of positive measure, the set `E/E` contains a neighborhood of `1`. -/
+@[to_additive "**Steinhaus Theorem** In any locally compact group `G` with a haar measure `μ`,
+  for any measurable set `E` of positive measure, the set `E - E` contains a neighborhood of `0`."]
+theorem steinhaus_theorem_mul (μ : measure G) [is_haar_measure μ] [locally_compact_space G]
+  (E : set G) (hE : measurable_set E) (hEpos : 0 < μ E) :
+  ∃ (U : set G) , U ∈ nhds (1 : G) ∧ U ⊆ E / E :=
+begin
+  rcases (exists_subset_measure_lt_top hE hEpos) with ⟨L, hL, hLE, hLpos, hLtop⟩,
+  have hK := measurable_set.exists_lt_is_compact_of_ne_top hL (ne_of_lt hLtop) hLpos,
+  rcases hK with ⟨K, hKL, hK, hKpos⟩,
+  have hKtop : μ K ≠ ⊤,
+  { apply ne_top_of_le_ne_top (ne_of_lt hLtop),
+    apply measure_mono hKL },
+  have h2 : μ K ≠ 0,
+  { exact ne_of_gt hKpos },
```

Comment:
These two lines could be
```suggestion
  have h2 : μ K ≠ 0 := hKpos.ne',
```
But, even better, remove it and just use `hKpos.ne'` whenever you need it: it will be just as clear.